### PR TITLE
Handle type helpers

### PR DESCRIPTION
**Pull request by circular17**
Hi!

This is a new attempt at implementing type helpers. The idea is that when encountering the `type` keyword, a forward check for the next token to be `helper` let us know if this is a type helper and not a type redefinition.

I was not sure how the unit tests work. Though I tried it on BGRABitmapTypes and it was successful with the code mentioned in issue #113.

Regards

### CODE REVIEW

**Comment by circular17** [comment]
In fact, it doesn't work with type redefinition. With the code:
```pascal
type
  TBGLTextureHandle = type Pointer; 
```

I get the error: `Cannot UnGet more than one token in TScanner.`

I will look into it.

**Comment by circular17** [comment]
Ok I think I've fixed the type redefinition.

I invite you to have a look.

Regards

File: source/component/PasDoc_Items.pas
Comment (michaliskambi) [inline source/component/PasDoc_Items.pas:758]: Let's call it `CIO_TYPEHELPER` (instead of just `CIO_TYPE`), to be more obvious. 

Then the comment `{helper only}` will also not be necessary -- the `CIO_TYPEHELPER` name already communicates everything :)

File: source/component/PasDoc_Items.pas
Comment (michaliskambi) [inline source/component/PasDoc_Items.pas:3063]: Let's make it `'type helper'`, not just `'type'`. 

Similar to previous comment, this makes it more obvious, and then there's no need for a comment `{helper only}` here.

File: source/component/PasDoc_Parser.pas
Comment (michaliskambi) [inline source/component/PasDoc_Parser.pas:2366]: The test `(t2.MyType = TOK_IDENTIFIER) and (t2.Info.StandardDirective = SD_HELPER)` can be done easier, `T2.IsStandardDirective(SD_HELPER)`. And then it looks concise enough to not even need an extra local variable `IsTypeHelper ` IMHO, so you can just test 

```
if T2.IsStandardDirective(SD_HELPER) then
....
```

P.S. Call it `t2` not `T2`. I know we are not consistent in PasDoc with identifier naming, but the guideline is to follow `PascalCase` for everything, like in CGE ( https://castle-engine.io/coding_conventions#pascal_case ).

File: source/component/PasDoc_Scanner.pas
Comment (michaliskambi) [inline source/component/PasDoc_Scanner.pas:78]: Remove this extra `{ TScanner }` comment. 

It was likely added by Lazarus Code Tools. We advise to turn off in Lazarus IDE the "Header comment for class", see also https://github.com/pasdoc/pasdoc/issues/149#issuecomment-1308837496 . This comment doesn't really add anything useful IMHO, and it obscures the important API documentation above (`{ This class scans....`).

**Review by michaliskambi — CHANGES_REQUESTED** [review]
Thank you a lot for the pull request!

1. I have made a few (rather local) requests for improvements below.

2. As for how to run tests -- I have enabled running tests automatically (by "GitHub Actions") for this PR. They've been blocked by default for security for 1st PRs from new contributors. So you don't need to do anything :), let's just wait for "GitHub Actions" results. Oh, and I see it already finished, and the report is "All checks have passed", excellent.

    In general, if you're curious, you can read `tests/README.md` on how to run the tests locally. You need to install some software and then do `make tests` from the command-line. But it is also absolutely OK to just leave this boring job to "GitHub Actions", let the system do it's job, without any hassle for us developers :)

3. I was a bit afraid initially that changing `FBufferedTokens` into a list (to hold multiple tokens) could be a performance hit, but it is not, according to my tests. And I see why you needed to do that -- we need a buffer to effectively "unget" 2 tokens ("type" and "helper") in case we have "type" followed by something else than "helper".

    So, no problem with this for me. FYI, these are quick test results:

    Times before (look only at difference in seconds between each step):

    ```
    00.19.20 Running tests for format html
    00.19.29 Running tests for format htmlhelp
    00.19.38 Running tests for format latex
    00.19.44 Running tests for format latex2rtf
    00.19.50 Running tests for format simplexml
    00.19.56 Running tests for format php
    ```

    Times after:

    ```
    00.23.36 Running tests for format html
    00.23.44 Running tests for format htmlhelp
    00.23.52 Running tests for format latex
    00.23.58 Running tests for format latex2rtf
    00.24.04 Running tests for format simplexml
    00.24.10 Running tests for format php
    ```

    I also tested processing CGE docs time (`make clean html` in https://github.com/castle-engine/castle-engine/tree/master/doc/pasdoc ). Times were practically equal (60 vs 61 seconds :), but each time can differs by more than 1 sec ~randomly).

    These were done by FPC 3.2.2 on Win64 now.

Summary:
- Everything is good. 
- AD 2 and AD 3 points above were just FYI, all is good.
- Please address AD 1 (my comments below) and then I'll be happy to merge this.

Thanks again!

File: source/component/PasDoc_Items.pas
Comment (circular17) [reply source/component/PasDoc_Items.pas:758]: Hi Michalis,

It's been some time. Sorry for the delay.

So, I don't think it would make sense to change it to CIO_TYPEHELPER because for the record helper, CIO_RECORD is used. Here it is the type that is targeted by the helper, not the helper itself.

Regards

File: source/component/PasDoc_Items.pas
Comment (circular17) [reply source/component/PasDoc_Items.pas:3063]: It is kind of the same remark here. "record" is the label used for "record helper". If we add "helper" here, then the helper label would become "type helper helper".

File: source/component/PasDoc_Parser.pas
Comment (circular17) [reply source/component/PasDoc_Parser.pas:2365]: Ok, applied the change.

File: source/component/PasDoc_Scanner.pas
Comment (circular17) [reply source/component/PasDoc_Scanner.pas]: Yes, that's probably an auto-comment. I've removed it.

Thanks for the link, I've updated the configuration. It is not useful and also it get in the way of the documentation generated by `pasdoc`.

**Comment by circular17** [comment]
Hi @michaliskambi,

Thanks for the details on testing, that's interesting. Indeed there is almost no time related to the undo list of tokens, because once the list is allocated, it just stays in memory for the rest of the process.

I've applied some changes. For the others, I've explained why I don't think it makes sense.

Looking forward to your response. I have another pull request to propose after this one is merged.

Warm regards

**Pull request merged**
